x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -480,37 +670,63 @@ protected PhysicalPlan rule(OrderExec plan) {
                     Missing missing = Missing.from(order.nullsPosition());
 
                     // check whether sorting is on an group (and thus nested agg) or field
-                    Attribute attr = ((NamedExpression) order.child()).toAttribute();
-                    // check whether there's an alias (occurs with scalar functions which are not named)
-                    attr = qContainer.aliases().getOrDefault(attr.id(), attr);
-                    GroupByKey group = qContainer.findGroupForAgg(attr);
+                    Expression orderExpression = order.child();
+
+                    // if it's a reference, get the target expression
+                    if (orderExpression instanceof ReferenceAttribute) {
+                        orderExpression = qContainer.aliases().get(orderExpression);
+                    }
+                    String lookup = Expressions.id(orderExpression);
+                    GroupByKey group = qContainer.findGroupForAgg(lookup);
 
+                    // TODO: handle score
```

Comment:
Should be fixed before merging the PR?

Reply:
It's already addressed later in the file (2nd else). Removed the line.